include/particle.tcc

```diff
@@ -439,6 +439,7 @@ bool mpm::Particle<Tdim, Tnphases>::update_volume_strainrate(unsigned phase,
       Eigen::VectorXd strain_rate_centroid =
           cell_->compute_strain_rate_centroid(phase);
       this->volume_(phase) *= (1. + dt * strain_rate_centroid.head(Tdim).sum());
+      this->mass_density_(phase) = this->mass_(phase) / this->volume(phase);
```

Comment:
For single point two phase formulation, I am not sure if this would work. Because, the water phase mass is not guaranteed accurate, so we cannot update density using mass_(phase).

Reply:
@bodhinandach Please include Tianchi and Weijiang to this discussion before making this change.

Reply:
@srhgk2 I will discuss with them. Thanks for pointing it out.

Reply:
Use strain rate (deformation gradient) to update density, if you want to update it.

Reply:
My motivation now is whether I should interpolate the mass_density_ too or not, cause I am not doing that through the hdf5

Reply:
@srhgk2 I modified the update as we discussed